aptos-move/aptos-vm/src/verifier/view_function.rs

```diff
@@ -56,6 +58,7 @@ pub(crate) fn validate_view_function(
     let allowed_structs = get_allowed_structs(struct_constructors_feature);
     let args = transaction_arg_validation::construct_args(
         session,
+        &mut UnmeteredGasMeter,
```

Comment:
View functions are typically not charged any gas on their own (unless called from some entry function). I am hoping the metering would not change that.

Reply:
previously construct_args where not directly accepting gas-meter as argument, but it was creating `UnmeteredGasMeter` internally to do the construction. But as long as we wanted to charge for `AutomationTransaction` embedded payload validation, I have updated the functions to accept gas-meter as input. So here specifying `UnbeteredGasMeter` does not affect affect original view-function economy and no charges will be applied.